bundles/org.openhab.binding.androiddebugbridge/src/main/java/org/openhab/binding/androiddebugbridge/internal/AndroidDebugBridgeHandler.java

```diff
@@ -0,0 +1,230 @@
+    public void initialize() {
+        var currentConfig = getConfigAs(AndroidDebugBridgeConfiguration.class);
+        config = currentConfig;
+        adbConnection.configure(currentConfig.ip, currentConfig.port);
+        updateStatus(ThingStatus.UNKNOWN);
+        connectionCheckerSchedule = scheduler.scheduleWithFixedDelay(this::checkConnection, 0,
+                currentConfig.refreshTime, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        var schedule = connectionCheckerSchedule;
+        if (schedule != null) {
+            schedule.cancel(true);
+            connectionCheckerSchedule = null;
+        }
+        adbConnection.disconnect();
+    }
+
+    public void checkConnection() {
+        var currentConfig = config;
+        if (currentConfig == null)
+            return;
+        logger.debug("Refresh device {} status", currentConfig.ip);
+        if (adbConnection.isConnected()) {
+            if (thing.getStatus() != ThingStatus.ONLINE)
+                updateStatus(ThingStatus.ONLINE);
+            refreshStatus();
+        } else {
+            if (thing.getStatus() != ThingStatus.OFFLINE) {
+                updateStatus(ThingStatus.OFFLINE);
+            }
+            try {
+                adbConnection.connect();
+            } catch (AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException e) {
+                logger.debug("Error connecting to device: {}", e.getMessage());
+                return;
+            }
+            if (adbConnection.isConnected()) {
+                updateStatus(ThingStatus.ONLINE);
+                refreshStatus();
+            }
+        }
+    }
+
+    private void refreshStatus() {
+        try {
+            handleCommandInternal(new ChannelUID(this.thing.getUID(), MEDIA_VOLUME_CHANNEL), RefreshType.REFRESH);
+        } catch (IOException | InterruptedException | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException
+                | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceReadException e) {
+            logger.warn("Unable to refresh media volume: {}", e.getMessage());
+        }
+        try {
+            handleCommandInternal(new ChannelUID(this.thing.getUID(), MEDIA_CONTROL_CHANNEL), RefreshType.REFRESH);
+        } catch (IOException | InterruptedException | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException
+                | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceReadException e) {
+            logger.warn("Unable to refresh play status: {}", e.getMessage());
+        }
+        try {
+            handleCommandInternal(new ChannelUID(this.thing.getUID(), CURRENT_PACKAGE_CHANNEL), RefreshType.REFRESH);
+        } catch (IOException | InterruptedException | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceException
+                | AndroidDebugBridgeDevice.AndroidDebugBridgeDeviceReadException e) {
+            logger.warn("Unable to refresh current package: {}", e.getMessage());
+        }
```

Comment:
InterruptExceptions should cause an immediate return from the current method. You should not perform additional operations once an interrupt has been received since you would be delaying openhab from shutting down.